crates/ursa-network/src/service.rs

```diff
@@ -196,6 +198,10 @@ pub struct UrsaService<S> {
     _pending_requests: HashMap<RequestId, ResponseChannel<UrsaExchangeResponse>>,
     /// Pending responses.
     pending_responses: HashMap<RequestId, oneshot::Sender<Result<UrsaExchangeResponse>>>,
+    /// Connected peers.
+    peers: HashSet<PeerId>,
```

Comment:
```rust
    /// Information about connected peers.
    peer_info: HashMap<PeerId, Vec<Multiaddr>>,
```

if are not using peer_info where are we strong the mapping of the `peer_id` and their information

Reply:
It's not needed anymore, all the behaviours track this themselves internally and can be accessed at any point with `behaviour.get_peer_addresses()`. This includes addresses from every behaviour internally and you can verify by running a `cargo expand` to view the macro provided implementation

Reply:
Yeah that make sense given i suggested the move away form having a custom handler still how do we handle custom peer info

Reply:
> custom peer info

I didn't see any actual usages of the peer_info struct other than insertions, so I don't think we need to worry for now. What are you anticipating we'd need to store?

Reply:
When we handle custom gossip sub logic with scoring, content routing, and more. You can leave it out, for now, I guess.